File: 백준/Silver/9613. GCD 합/GCD 합.py
t = int(input())


def gcd(a, b):
    while b > 0:
        a, b = b, a % b
    return a


def comb(d, idx, arr):
    global ans
    if d == 2:
        ans += gcd(max(arr), min(arr))
        return
    for i in range(idx, n):
        comb(d+1, i+1, arr+[gcds[i]])
    return


for _ in range(t):
    arr = list(map(int, input().split()))
    n = arr[0]
    gcds = arr[1:]
    ans = 0
    comb(0, 0, [])
    print(ans)

File: 백준/Silver/9613. GCD 합/GCD 합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

  static int GCD(int a, int b) {
    int tmp = 0;
    while (b != 0) {
      tmp = a;
      a = b;
      b = tmp % b;
    }
    return a;
  }

  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    int t = Integer.parseInt(br.readLine());
    for (int T = 0; T < t; T++) {
      String[] nums = br.readLine().split(" ");
      long ans = 0L;

      for (int i = 1; i < nums.length; i++) {
        for (int j = i + 1; j < nums.length; j++) {
          int a = Integer.parseInt(nums[i]), b = Integer.parseInt(nums[j]);
          long gcd = GCD(Math.max(a, b), Math.min(a, b));
          ans += gcd;
        }
      }

      System.out.println(ans);
    }

  }
}
